src/dodal/devices/shutter.py

```diff
@@ -0,0 +1,27 @@
+from enum import Enum
+
+from bluesky.protocols import Movable
+from ophyd_async.core import AsyncStatus, StandardReadable
+from ophyd_async.epics.signal import epics_signal_rw
+
+
+class OpenState(Enum):
+    CLOSE = 0
+    OPEN = 1
+
+
+class Shutter(StandardReadable, Movable):
+    def __init__(self, prefix: str, name: str):
+        self.position = epics_signal_rw(
+            write_pv=prefix + "CTRL2", read_pv=prefix + "STA", datatype=int, name=name
+        )
+        super().__init__(
+            name=name,
+        )
+        self.set_readable_signals([self.position])
+
+    def set(self, open_val: int | OpenState):
+        if isinstance(open_val, OpenState):
+            open_val = open_val.value
+        shutter_status: AsyncStatus = self.position.set(open_val)
+        return shutter_status
```

Comment:
Can't `self.position` simply be a signal of type `OpenState`?

Reply:
could be

Reply:
It could even be 

```python
self.open = epics_signal_rw(
            write_pv=prefix + "CTRL2", read_pv=prefix + "STA", datatype=bool, name=name
        )
```

Reply:
I'm not sure I understand the case for simplicity here.

In my model of this, I require the 'set' method to override the device 'set' method. 

I agree that OpenState is a bit of a redundant thing.


But I'd leave it at int as bool is less flexible and IDK how does it go throguh epics channels

Reply:
Less flexible is good, the shutter has literally 2 states. A bool is the perfect representation of that.

here https://github.com/bluesky/ophyd-async/blob/f6f9cc0762f8c4a9c6555a3734799b0269ae266c/src/ophyd_async/epics/_backend/_p4p.py#L203
and here https://github.com/bluesky/ophyd-async/blob/f6f9cc0762f8c4a9c6555a3734799b0269ae266c/src/ophyd_async/epics/_backend/_aioca.py#L128
are where bool is handled.

The signal should either be an Enum or a bool, I am insistent on that. int is just incorrect.

Reply:
dropped

Reply:
Going to unresolve this one as I still think it should be a signal. That would also make the tests easier as you can use `set_sim_value` to mock the state and `get_value` rather than `read`

Reply:
@stan-dot https://github.com/DiamondLightSource/dodal/blob/main/src/dodal/devices/undulator.py

These are signals, you shouldn't have to override `set` on the class inheriting from `StandardReadable`, you should just be able to add signals to it.

Reply:
done